### PR TITLE
Prow endpoints monitoring(1123)

### LINKED ISSUE #1123: add status checks for health endpoints
**Description**
Add another grafana dashboard to display health endpoint data for us to better monitor component health. 

**Reasons**
Current Grafana dashboards are not covering health endpoints of components and we're not notified if a component is not working (was not working).

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

- Add alertmanager configuration for prow slack notification channel.
- Add prow notification template.
- Add prow services and endpoints alerting rules
- Add prow service and endpoints charts to grafana Prow dashboard.
- Split alertmanager, prometheus and grafana settings in to separate yaml files.
- Update monitoring readme.
- Update prow installation scripts to enable k8s client cert authentication.
- Update installation scripts to detect if env variables are set.
- Automation additional installation steps.

**Related issue(s)**
`Resolves #1123`

### CODE REVIEW

File: docs/prow/prow-monitoring.md [hunk 1]
```diff
@@ -81,3 +81,12 @@ Follow these steps to save the dashboard:
    ```
 
    > **NOTE:** `--recreate-pods` is required because the Secret with the Grafana password is regenerated during the upgrade and it needs to be populated to Grafana.
+
+## Add recording and alerting rules
+
+1. Add new recording or alerting rules to the prometheusrule spec [`prow_prometheusrules.yaml`](../../prow/cluster/resources/monitoring/templates/prow_prometheusrules.yaml).
```
Comment (klaudiagrz) [inline docs/prow/prow-monitoring.md]:
```suggestion
1. Add new recording or alerting rules to the [PrometheusRule specification](../../prow/cluster/resources/monitoring/templates/prow_prometheusrules.yaml).
```

File: docs/prow/prow-monitoring.md [hunk 2]
```diff
@@ -81,3 +81,12 @@ Follow these steps to save the dashboard:
...
+
+2. Replace existing prometheusrule object with current file version.
```
Comment (klaudiagrz) [inline docs/prow/prow-monitoring.md]:
```suggestion
2. Replace the existing PrometheusRule object with the current file version.
```

File: development/install-prow.sh [hunk 3]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
 
+
+######################################
```
[the remark this answers is not in the document]
Comment (dekiel) [reply development/install-prow.sh]: We need to search for references before file can be removed. This has to be done later.

File: development/create-service-accounts-for-fork.sh [hunk 4]
```diff
@@ -63,6 +67,8 @@ SA_REAL_NAME=sa-gcs-plank
 SA_NAME=$(echo ${SA_REAL_NAME} | cut -c1-19)-$SHORT_IDENTIFIER
 SA_DISPLAY_NAME=$SA_REAL_NAME-$SHORT_IDENTIFIER
 SECRET_FILE=$SECRET_FOLDER/$SA_REAL_NAME
+#SA_NAME=${SA_REAL_NAME}
```
[the remark this answers is not in the document]
Comment (dekiel) [reply development/create-service-accounts-for-fork.sh]: Changes in prow installation are side effect of changes in monitoring. I will remove comments during work on installation.

File: prow/cluster/resources/monitoring/templates/prow_prometheusrules.yaml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                for: 30s
+                labels:
+                    prowservice: deck
+                    app: prow
+                    severity: critical
+                annotations:
+                    summary: Service not present.
+                    description: Deck service is not present on Prow k8s cluster.
+              #TIDE
+              - alert: ProwTideServiceNotPresent
```
[the remark this answers is not in the document]
Comment (dekiel) [reply prow/cluster/resources/monitoring/templates/prow_prometheusrules.yaml]: Yes, this will work as tide service and endpoints are present on cluster.

**Pull request merged**